File: src/main/java/in/neuw/resource/demo/models/ErrorDto.java
package in.neuw.resource.demo.models;

import lombok.Getter;
import lombok.Setter;
import lombok.experimental.Accessors;
import org.springframework.http.HttpStatus;

/**
 * @author Karanbir Singh on 04/21/2020
 */
@Getter
@Setter
@Accessors(chain = true)
public class ErrorDto {

    private String message;
    private HttpStatus status = HttpStatus.INTERNAL_SERVER_ERROR;
    private int code;
    private String error;

    public ErrorDto(String message, HttpStatus status) {
        this.message = message;
        this.status = status;
    }

    public int getCode() {
        return status.value() * 1000 + code;
    }

}

File: src/main/kotlin/in/neuw/resource/demo/models/ErrorDto.kt
package `in`.neuw.resource.demo.models

import org.springframework.http.HttpStatus

data class ErrorDto(val message: String?, val status: HttpStatus = HttpStatus.INTERNAL_SERVER_ERROR) {

    var code: Int = this.status.value() * 1000 + 100
        get() = this.status.value() * 1000 + field

    var error: String = ""

}